scrapy_poet/injection.py

```diff
@@ -106,6 +109,7 @@ def build_plan(self, request: Request) -> andi.Plan:
             callback,
             is_injectable=is_injectable,
             externally_provided=self.is_class_provided_by_any_provider,
+            overrides=self.overrides_registry.overrides_for(request).get
```

Comment:
What do you think about changing overrides_for interface to return functions? This is a low-level API, so maybe we should be leaning towards power / completeness vs ease of use here.

Reply:
I reckon it would be worth even if it is done only for the fact that the return annotation becomes `Callable[[Callable], Callable]` :rofl: 

Now seriously, the problem I see is that the interface is then very convoluted. See this example of a simple custom registry in such a case:

```python
            class MyRegistry(OverridesRegistryBase):

                def overrides_for(self, request):
                    def overrides_fn(request, cls):
                        if get_domain(request.url) == 'toscrape.com':
                            return BSTBookItem if cls is BookItem else return None
                        return None

                    return partial(overrides_fn, request)
```

The flexibility advantage that you could get makes sense if you have families of things to overrides and you don't want to be explicit about their mapping. For example:

```python
                def overrides_for(self, request):
                    def overrides_fn(request, cls):
                        if issubclass(cls, BookPage):
                            return xxx

                    return partial(overrides_fn, request)
```

But I cannot imagine a real example where this makes sense at all. So my vote here is not to over-engineer it and leave it at is. But I have a open mind :smile:, if you have an example where this could be useful, please, tell me.